mmcls/datasets/voc.py

```diff
@@ -53,16 +88,19 @@ def load_annotations(self):
                 else:
                     labels.append(label)
 
-            gt_label = np.zeros(len(self.CLASSES))
             # The order cannot be swapped for the case where multiple objects
             # of the same kind exist and some are difficult.
-            gt_label[labels_difficult] = -1
-            gt_label[labels] = 1
+            all_gt_labels[index, labels_difficult] = -1
+            all_gt_labels[index, labels] = 1
 
-            info = dict(
+            sample = dict(
                 img_prefix=self.data_prefix,
                 img_info=dict(filename=filename),
-                gt_label=gt_label.astype(np.int8))
-            data_infos.append(info)
+                gt_label_index=index)
```

Comment:
Yes, this is not beneficial to reducing mem cost, but I still need this to deal with "insufficient-file-descriptors" problem.

That is, because of the default sharing strategy `file_descriptor` (see section "Sharing strategies" in the [doc](https://pytorch.org/docs/stable/multiprocessing.html) or the following figure), sharing large number of tensors will likely exhaust file handles. That means if you have `N` training samples, then you should always set your `ulimit -n` bigger than `N`, that's not realistic for large datasets. Thus I stack `gt_label`s to a single tensor `all_gt_label` to avoid that. The problem exists and this solution works, I have verified them. Another possible solution is to change sharing strategy to `file_system` but I'm not sure whether it brings other side effects so I didn't change that.

![capture](https://user-images.githubusercontent.com/31942171/128799947-da738eb2-4455-42ba-8b31-2755ed3c3ea4.jpg)